allennlp/data/token_indexers/wordpiece_indexer.py

```diff
@@ -108,8 +112,10 @@ def tokens_to_indices(self,
         offset = len(wordpiece_ids) if self.use_starting_offsets else len(wordpiece_ids) - 1
 
         for token in tokens:
+            # Lowercase if necessary
+            text = token.text.lower() if self._do_lowercase else token.text
```

Comment:
One ugly workaround (for my other comment) would be to have a short list of exception tokens here which don't get lower cased (e.g., everything in square brackets, or [PAD], [UNK], [CLS], [SEP], [MASK])

Reply:
yeah, I'll add something like this